client/src/site-search/search-results.tsx

```diff
@@ -84,6 +87,82 @@ class ServerOperationalError extends Error {
   }
 }
 
+async function searchWithAPI(url, ga) {
+  const response = await fetch(url);
+  if (response.status === 400) {
+    const badRequest = await response.json();
+    throw new BadRequestError(badRequest.errors);
+  } else if (response.status >= 500) {
+    throw new ServerOperationalError(response.status);
+  } else if (!response.ok) {
+    throw new Error(`${response.status} on ${url}`);
+  }
+
+  // See docs/experiments/0001_site-search-x-cache.md
+  const xCacheHeaderValue = response.headers.get("x-cache");
+  ga("send", {
+    hitType: "event",
+    eventCategory: "Site-search X-Cache",
+    eventAction: url,
+    eventLabel: xCacheHeaderValue || "no value",
+  });
+
+  return await response.json();
+}
+
+async function searchWithPagefind(params) {
+  const startTime = new Date().getTime();
+  const locale = params.get("locale").toLowerCase();
+  // eval stops {webpack/react/typescript/g-d knows what else}
+  // from messing with this or complaining that the module doesn't exist.
+  // eslint-disable-next-line no-eval
+  const pagefind = await eval(
+    // eslint-disable-next-line no-template-curly-in-string
+    "import(`/static/js/pagefind/${locale}/pagefind.js`)"
+  );
+  const search = await pagefind.search(params.get("q"));
+  const resultsPerPage = 10;
+  let page = parseInt(params.get("page"));
+  if (page <= 0 || isNaN(page)) page = 1;
+  const startIndex = (page - 1) * resultsPerPage;
+  const results =
+    startIndex >= 0
+      ? await Promise.all(
+          search.results
+            .slice(startIndex, startIndex + resultsPerPage)
+            .map((r) => r.data())
+        )
+      : [];
+  const documents = results.map((result) => {
+    return {
+      mdn_url: `/${locale}/docs${result.url}`,
+      title: result.meta.title,
+      // note: summary is unused when highlight is non-empty
+      // and we don't have a good way of getting a summary
+      // without doing quite a bit of work
+      // (in particular, result.meta.content has HTML tags)
```

Comment:
Maybe mark this comment explicitly as TODO, since we would probably want to get around to that at some point.

Reply:
it would be completely useless unless the `Results` function is changed in the future... i'm just not sure it's worth the effort of figuring out which HTML tags contain text, etc.

Reply:
Ah - my bad, think I misunderstood what that field was for.